File: julia/linecounter.jl
function line_counter(file)
	if occursin(".gz", file)
		GZip.open(file) do f
			line=0
			for i in eachline(f)
				if !occursin("#",i)
					line+=1
				end
			end
			return line
		end
	else
		open(file) do f
			line=0
			for i in eachline(f)
				if !occursin("#",i)
					line+=1
				end
			end
			return line
		end
	end
end
println(string(line_counter(ARGS[1]))," lines")

File: python/line_counter.py
def lineCounter(file):
	import gzip
	import io
	cnt=0
	if '.gz' in file:
		with io.TextIOWrapper(io.BufferedReader(gzip.open(file))) as fh:
			for line in fh:
				if '#' not in line:
					cnt+=1
	else:
		with open(file) as fh:
			for line in fh:
				if '#' not in line:
					cnt+=1
	print(str(cnt)+' lines')
if __name__ == '__main__':
	file = input('What file? ')
	import os
	import sys
	file=file.strip()
	if os.path.exists(file) == False:
		print('barnacles! File not found, try again')
		sys.exit()
	lineCounter(file)

